backend/src/api/services/question/question.service.ts

```diff
@@ -46,14 +46,27 @@ export const getSidebarMetadata = async (studentID: string) => {
 };
 
 export const updateUserBookmark = async (studentID: string, questionID: string) => {
-    const students: [StudentProgress] = await loadData('progress.json');
-    const student = students.find(student => student.studentID === studentID);
-    if (!student) {
-        throw new Error('Student not found');
-    }
-    student.bookMark.push(questionID);
+    try {
+        // Check if the question exists
+        const question = await QuestionBaseModel.findById(questionID);
+        if (!question) {
+            throw new Error('Question not found');
+        }
```

Comment:
Instead of throwing an error, we should throw a HttpError for the bad response.
```
throw new HttpError(status.NOT_FOUND, `question is not found with question ID ${questionID}` );
```

Reply:
HttpError is our custom error that will be caught by our custom error middleware.